lib/features/sks-menu/data/repository/sks_menu_repository.dart

```diff
@@ -0,0 +1,29 @@
+import "package:collection/collection.dart";
+import "package:flutter_riverpod/flutter_riverpod.dart";
+import "package:riverpod_annotation/riverpod_annotation.dart";
+
+import "../../../sks_people_live/data/api_client/sks_client.dart";
+import "../api_client/sks_menu_client.dart";
+import "../models/dish_category_enum.dart";
+import "../models/sks_menu_data.dart";
+
+part "sks_menu_repository.g.dart";
+
+@riverpod
+Future<Map<DishCategory, List<SksMenuDish>>> getSksMenuData(Ref ref) async {
```

Comment:
Have you considered `IMap` type from immutable collections?

Reply:
Good point! I forgot about that